api-lib/make-telegram-event.ts

```diff
@@ -0,0 +1,29 @@
+import type { VercelRequest, VercelResponse } from '@vercel/node';
+
+import { GraphQLTypes } from '../src/lib/gql/zeusHasuraAdmin';
+
+import { EventTriggerPayload, OperationTypes } from './types';
+
+export default function makeTelegramEvent<
+  X extends keyof GraphQLTypes,
+  Y extends OperationTypes
+>(
+  msgHandler: (
+    payload: EventTriggerPayload<X, Y>,
+    channels: { discord?: boolean; telegram?: boolean }
+  ) => Promise<boolean>
+) {
+  return async function handler(req: VercelRequest, res: VercelResponse) {
+    try {
+      const sent = await msgHandler(req.body, { telegram: true });
+      res
+        .status(200)
+        .json({ message: `Telegram message ${sent ? 'sent' : 'not sent'}` });
+    } catch (e) {
+      res.status(500).json({
```

Comment:
Eventually we should move all these res.status(x).json() errors to use `ErrorResponse` or `ErrorResponseWithStatus`

Reply:
Yeah theres a good amount of cleanup to do